### ISSUE TITLE
Trouble running QSHPYRUN in batch job

### ISSUE THREAD

**Issue by edward-vo-eog** [richardschoen/QshOni#19]
I ran this command: 

SBMJOB CMD(QSHONI/QSHPYRUN SCRIPTDIR('/home/evo/python') SCR 
           IPTFILE(gldtpf.py) DSPSTDOUT(*YES) LOGSTDOUT(*YES 
           ) PRTSTDOUT(*YES) DLTSTDOUT(*NO) IFSSTDOUT(*YES)  
           IFSFILE(GLDTPF_LOG.TXT))                          
       JOB(GLDTPF_PY) 

The spool file containing the error is in the attached PDF. I've been using this command through batch job and this issue recently popped up without me changing the approach

[QPJOBLOG_EVO_GLDTPF_PY_785966_1.pdf](https://github.com/richardschoen/QshOni/files/10922906/QPJOBLOG_EVO_GLDTPF_PY_785966_1.pdf)

**Comment by richardschoen** [comment]
Looks to me like you need to specify the IFSFILE full path when outputting to the IFS: 
``` IFSFILE(GLDTPF_LOG.TXT)``` 
should be something like  
```IFSFILE('/tmp/GLDTPF_LOG.txt')```

Also if outputting to an IFS file, perhaps set the other stdout options to *NO for DSPSTDOUT, PRTSTDOUT, and LOGSTDOUT.

**Comment by richardschoen** [comment]
Also QSHPATHLOC data area should be a *char value: 
```
                              Display Data Area                           
                                                            System:   SYS1
Data area . . . . . . . :   QSHPATHLOC                                    
  Library . . . . . . . :     QSHONI                                      
Type  . . . . . . . . . :   *CHAR                                         
Length  . . . . . . . . :   10                                            
Text  . . . . . . . . . :   Path position for open source packages        
                                                                          
           Value                                                          
Offset      *...+....1....+....2....+....3....+....4....+....5            
    0      '*BEGIN    '                                                   
                                                                          
   ```

**Comment by richardschoen** [comment]
Here is a copy of the suspicious error:

```
CPF0811 Escape 40 03/08/23 10:28:02.693783 QCLRTVDA QSYS 0285 QSHPATHC QSHONI 0012
 Message . . . . : RTNVAR parameter has incorrect length for data area.
 Cause . . . . . : The program attempted to retrieve the contents of data
 area QSHPATHLOC in library QSHONI into a program variable which has an
 incorrect length. If the type of RTNVAR is *LGL, the data area or the
 substring of the data area specified does not have a length of 1. If the
 type of RTNVAR is *CHAR, it is not as long as the data area or substring
 specified. If the type of RTNVAR is *DEC, it is not long enough to hold the
 data in the specified data area. Recovery . . . : Determine the data area
 size and contents by displaying the data area (DSPDTAARA command). Then do
 one of the following and try the request again: -- Change the size of either
 the RTNVAR parameter or the data area. -- Change the substring length. --
 Change the contents of the data area.
CPF9898 Escape 40 03/08/23 10:28:02.693880 QSHPATHC QSHONI 0066 QSHEXECC
```

**Comment by edward-vo-eog** [comment]
I checked the QSHPATHLOC data area and it is in fact type *CHAR (with length 18), and tried running with the DSPSTDOUT, PRTSTDOUT, and LOGSTDOUT to *NO and fixed the IFS file variable to include full path, and still get the same error. I've attached the spool file but looks like it's the same error.

My new command: QSHONI/QSPYRUN SCRIPTDIR('/Python_Dev/PY_DEV/Ed') SCRIPTFILE(GLDTPF.py) PYVERSION(3.6)

I've tested with different Python programs on DB2 that runs fine interactively without using QSHPYRUN. I also tested running them with PYONI/PYRUN command and it also works fine.
[](url)
[ACS.splf-11087682629927865608.pdf](https://github.com/richardschoen/QshOni/files/10984069/ACS.splf-11087682629927865608.pdf)
